src/styles/utils.scss

```diff
@@ -0,0 +1,23 @@
+@each $last-day in $month-length {
+  .calendar--month-length--#{$last-day} {
+    @for $days from 1 through $last-day {
```

Comment:
do not use `@for` inside `@each` - you generate ~120 similar styles (when 31 is enough)

add content and other styles you needed for `.calendar .day:nth-child` instead of `.calendar--month-length--x .day:nth-child`

and add `display: none` for days that are bigger than $month-length